lib/irc/ConnectionInstance.js

```diff
@@ -256,7 +257,7 @@ ConnectionInstance.create = function(server, opts, onCreatedCallback) {
 
     var connAttempts = 0;
     function retryForever() {
-        retryConnection().then(returnClient, function(err) {
+        let reschedule = function(err) {
```

Comment:
Please call this something different. It's really confusing that this is called `reschedule` AND there is `Scheduler.reschedule` AND we're passing functions around.